pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -57,13 +61,25 @@ def __init__(
                 f" Move your implementation to `{self.__class__.__name__}.teardown()` instead."
             )
 
+    @property
+    def accelerator(self) -> "pl.Accelerator":
+        return self._accelerator
+
+    @accelerator.setter
+    def accelerator(self, accelerator: "pl.Accelerator") -> None:
+        self._accelerator = accelerator
+
     @property
     def checkpoint_io(self) -> CheckpointIO:
         return self._checkpoint_io
 
     @property
     def precision_plugin(self) -> PrecisionPlugin:
-        return self._precision_plugin
+        return self._precision_plugin if self._precision_plugin is not None else PrecisionPlugin()
+
+    @precision_plugin.setter
+    def precision_plugin(self, precision_plugin: Optional[PrecisionPlugin]) -> None:
+        self._precision_plugin = precision_plugin
```

Comment:
Adding this in #11071

Reply:
Thank you @carmocca, will rebase if 11071 merge first. The accelerator_connector logic change in this PR need precision setters for now